### ISSUE TITLE
Exclude unmatched word forms from results

### ISSUE THREAD

**Issue by dveselov** [natasha/yargy#3]
When matching text with `gnc-match` (or similar labels) it's possible to do some sort of morphological disambiguation solving, for example, with given grammar:
```python
Grammar('Firstname_and_Lastname', [
    {
        'labels': [
            ('gram', 'Name'),
        ],
    },
    {
        'labels: [
            ('gram', 'Surn'),
            ('gnc-match', -1),
        ],
    },
]
```
Returned matches will contain tokens that have specified grammemes (`Name` and `Surn`) with `forms` attribute that contains all of word forms returned by pymorphy2, but it may be a significant to reduce forms only to that matched labels.

**Comment by dveselov** [comment]
Implemented for `gnc_match` label in 0af03bca040c154dfebc7b4d68026527d5b6c0ac

**Comment by dveselov** [comment]
Implemented for `number_match`, `case_match` and `gender_match` labels in e0f2a63cc7a46970a10ff5e9cc0c36b1346b5ee0
